tests/test_gradient.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import pytest
+from numpy import linalg
+
+from examples.Acoustic_codegen import Acoustic_cg
+from examples.containers import IGrid, IShot
+
+
+class Test_Gradient(object):
```

Comment:
One shouldn't use `_` for class names -- I *think* it's quite a common rule in Python